src/globals.c

```diff
@@ -126,12 +129,13 @@ ir_instr_t *add_instr(opcode_t op)
     return ii;
 }
 
-block_t *add_block(block_t *parent, func_t *func)
+block_t *add_block(block_t *parent, func_t *func, macro_t *macro)
```

Comment:
This change is to indicate that the block can access variables from function and both macro parameter declarations. I think it makes the semantics clearer than passing the pointer of macro from parent.

Reply:
Describe the design decisions in git commit messages as well.